### PR TITLE
RemoveNonExistingVarAnnotationRector: Allow return annotations

### LINKED ISSUE #7859: RemoveNonExistingVarAnnotationRector removes return-@var annotations
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/e83ea115-35d4-4945-ba71-c5a66af05ce2

```php
<?php declare(strict_types = 1);

class HelloWorld {
	public function getValue(): int|false|null {
		/** @var int|false|null */
		return filter_input(INPUT_GET, 'value', FILTER_VALIDATE_INT);
	}
}
```

### Responsible rules

* `RemoveNonExistingVarAnnotationRector`

## Expected Behavior

The `@var` annotation should not be removed, it is used to annotate the value being returned.

This syntax is understood by [PHPStan](https://phpstan.org/r/7383c016-5d87-468e-88e7-756480272f61), [Psalm](https://psalm.dev/r/0d6f65166b) and PhpStorm.

### PR DESCRIPTION
**Pull request by jlherren**
Fixes rectorphp/rector#7859

### CODE REVIEW

File: rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc]: please change file name to something like: `skip_direct_return_func_call.php.inc`

File: rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Rector\Tests\DeadCode\Rector\Node\RemoveNonExistingVarAnnotationRector\Fixture;
+
+class ReturnAnnotation
```
Comment (samsonasik) [inline rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc]:
```suggestion
class SkipDirectReturnFuncCall
```

File: rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc (hunk 1 above)
Comment (jlherren) [reply rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc]: I assume the other files are fine. I can't really tell a pattern with the non-skip files.

Comment (samsonasik) [reply rules-tests/DeadCode/Rector/Node/RemoveNonExistingVarAnnotationRector/Fixture/return_annotation.php.inc]: Yes, using skip_ prefix is for consistency

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @jlherren

**Pull request merged**